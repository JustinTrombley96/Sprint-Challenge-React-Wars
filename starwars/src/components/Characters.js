import React from 'react'
import Character from './Character'

const Characters = props => {
    return (
        <div>
            {console.log(props.characters)}
            {props.characters.map(character => <Character character={character} />)}
        </div>
    )
}

export default Characters