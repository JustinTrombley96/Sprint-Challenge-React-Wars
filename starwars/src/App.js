import React, { useState, useEffect} from 'react';
import axios from 'axios'
import './App.css';
import Characters from './components/Characters'
import { Box, borders } from '@material-ui/core'

const App = () => {
  // Try to think through what state you'll need for this app before starting. Then build out
  // the state properties here.

  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a 
  // side effect in a component, you want to think about which state and/or props it should
  // sync up with, if any.

  const [characters, setCharacters] = useState([])
  useEffect(() => {
    axios 
    .get('https://swapi.co/api/people/')
    .then(res => {
      setCharacters(res.data.results)
    })
    .catch(err => console.log(err))
  }, [])

  return (
    <div className="App">
    <Box bgcolor="primary.main" border={1} borderRadius="50%" borderColor="text.primary">
      <h1 className="Header">React Wars</h1>
      </Box>
      <Characters characters={characters} />
    </div>
  );
}

export default App;
