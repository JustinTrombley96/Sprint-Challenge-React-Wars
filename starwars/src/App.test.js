import React from 'react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import App from './App'
import Characters from './components/Characters';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  });

  it('matches snapshot 2', () => {
    const trees = renderer.create(<Characters />).toJSON()
    expect(trees).toMatchSnapshot()
  });