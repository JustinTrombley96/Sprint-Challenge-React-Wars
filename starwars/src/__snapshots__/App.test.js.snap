// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="Header"
  >
    React Wars
  </h1>
  <div />
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="App"
>
  <h1
    className="Header"
  >
    React Wars
  </h1>
  <div />
</div>
`;
